.circleci/config.yml

```
@@ -13,12 +13,92 @@ jobs:
       - checkout
       - run: make check
 
+  setup:
+    docker:
+      - image: mattermost/mattermost-build-webapp:oct-2-2018
+    working_directory: ~/go/src/github.com/mattermost/
+    steps:
+      - run:
+          name: Checkout mattermost-server
+          command: |
+            mkdir -p ~/.ssh/
+            echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config
+            git clone git@github.com:mattermost/mattermost-server.git
+
+            cd mattermost-server
+            git checkout $CIRCLE_BRANCH || git checkout master
+      - persist_to_workspace:
+          root: ~/go/src/github.com/mattermost
+          paths:
+            - mattermost-server
+
   test:
-    executor:
-      name: default
+    machine:
+      image: ubuntu-1604:201903-01
+    working_directory: ~/go/src/github.com/mattermost/mmctl
+    environment:
+      COMPOSE_PROJECT_NAME: "circleci"
     steps:
       - checkout
-      - run: make test
+      - attach_workspace:
+          at: ~/go/src/github.com/mattermost/
+      - run:
+          name: Run Docker compose
+          command: |
+            cd ../mattermost-server/build
+            docker-compose --no-ansi run --rm start_dependencies
+            cat ../tests/test-data.ldif | docker-compose --no-ansi exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest';
+            docker-compose --no-ansi exec -T minio sh -c 'mkdir -p /data/mattermost-test';
+            docker-compose --no-ansi ps
+          background: true
+      - run:
+          name: Wait for docker compose
+          command: |
+              sleep 5
+              docker run --rm --net circleci_mm-test appropriate/curl:latest sh -c "until curl --max-time 5 --output - http://mysql:3306; do echo waiting for mysql; sleep 5; done;"
+              docker run --rm --net circleci_mm-test appropriate/curl:latest sh -c "until curl --max-time 5 --output - http://elasticsearch:9200; do echo waiting for elasticsearch; sleep 5; done;"
+      - run:
+          name: Start Mattermost
+          command: |
+            ulimit -n 8096
+            cd ~/go/src/github.com/mattermost/mattermost-server
+
+            docker run --net circleci_mm-test \
+              --name mattermost_circleci \
+              --env MM_CONFIG='mysql://mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8' \
+              --env MM_TEAMSETTINGS_MAXUSERSPERTEAM=500 \
+              -d \
+              mattermost/mattermost-enterprise-edition:5.14.0-rc4
```

```suggestion
              mattermost/mattermost-enterprise-edition:master
```
will get the latest master available

---

or do you want to test only on 5.14 release?

---

Nope, that's definitely a good catch!